staging/src/k8s.io/apiserver/pkg/endpoints/filters/trace.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package filters
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	utiltrace "k8s.io/utils/trace"
+
+	"k8s.io/apiserver/pkg/endpoints/internal"
+	genericapirequest "k8s.io/apiserver/pkg/endpoints/request"
+)
+
+// WithTrace decorates a http.Handler with tracing for all the non-long running
+// requests coming to the server.
+func WithTrace(handler http.Handler, longRunningCheck genericapirequest.LongRunningRequestCheck) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		isLongRunning := false
+		if longRunningCheck != nil {
+			ri, ok := genericapirequest.RequestInfoFrom(req.Context())
+			if ok && longRunningCheck(req, ri) {
+				isLongRunning = true
+			}
+		}
+
+		if !isLongRunning {
+			ctx, trace := genericapirequest.WithTrace(req.Context(), "HTTP "+req.Method,
+				utiltrace.Field{Key: "url", Value: req.URL.Path},
```

Comment:
we don't need a field for the req.Method for query purposes.

Reply:
That would be much better. Adding.